src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -510,6 +510,14 @@ void ResolveContext::OnNewInterface(uint32_t interfaceId, const char * fullname,
     // resolving.
     interface.fullyQualifiedDomainName = hostnameWithDomain;
 
+    // DNSServiceResolve caches old entries with the same interface ID. This means that when the broadcast with the
+    // same interface has been updated then it will not be reflected in the list of interfaces. In this case, remove
+    // the existing entry in the map since the resolve seems to store it in sequential order and the last entry is
+    // the most up to date.
```

Comment:
I have no idea whether any of that ordering bit there is guaranteed.  @Abhayakara @StuartCheshire 

That said, what does "broadcast with the same interface has been updated" mean?  What exactly changed on the advertiser side?  What packets did it send to update the dns-sd information?

Reply:
Here's what i'm seeing 

```2023-02-23 15:14:48.630394-0800 TvCasting[12027:2767414] [all] 🔵 [1677194088630] [12027:2767414] CHIP: [DIS] Browsing for: _matterd._udp,_T35 at interface 0
2023-02-23 15:14:48.637278-0800 TvCasting[12027:2767414] [all] 🟢 [1677194088637] [12027:2767414] CHIP: [DIS] Mdns: OnBrowseAdd  name: D49A7571551947A4, type: _matterd._udp, domain: local., interface: 16
2023-02-23 15:14:48.637862-0800 TvCasting[12027:2767414] [all] 🟢 [1677194088637] [12027:2767414] CHIP: [DIS] Resolve type=_matterd._udp name=D49A7571551947A4 interface=16
2023-02-23 15:14:48.643144-0800 TvCasting[12027:2767579] [all] 🟢 [1677194088643] [12027:2767579] CHIP: [DIS] Mdns : OnNewInterface hostname:FAE9D8AB693A.local. fullname:D49A7571551947A4._matterd._udp.local. interface: 16 port: 5550 TXT:"VP=65521+32769,DT=35,DN=Cliff's\x20one"
2023-02-23 15:14:48.643386-0800 TvCasting[12027:2767579] [all] 🟢 [1677194088643] [12027:2767579] CHIP: [DIS] Mdns : OnNewInterface hostname:FAE9D8AB693A.local. fullname:D49A7571551947A4._matterd._udp.local. interface: 16 port: 5550 TXT:"VP=65521+32769,DT=35,DN=Cliff's\x20two"

Each time the DN value is being changed with the new attribute and the advertiser is sending a `goodbye / TTL=0` packet as it is changing the dns broadcast (does DNSServiceResolve respect these?).

Reply:
Ordering isn't guaranteed. If you have five answers, try all five. Don't just try the one you received last. mDNS and DNS are cache maintenance protocols, not a query/response protocols, although it can appear as if they are the latter.

Also, I hope that once you've gotten your answer from Resolve, you stop that resolve. You can leave browses going, but not Resolves. I'm realizing I may have spoken in a confusing way about this in the past—until you get an answer, definitely leave the Resolve going, but once you have your answer, it's done. If you want to track the state of the TXT and PTR records you need to do a query—in mDNSResponder this would be DNSServiceQueryRecord() as opposed to DNSServiceResolve().

Reply:
Of course, in principle there should only ever be one answer for TXT and SRV in the cache at the time, and we do try our best to make sure that it's the most recent answer. So I don't mean keep around stale data for hours and keep trying with it. If it doesn't work, that's when you should try to do another resolve.

But if you get several SRV records back in a Resolve query, all but the last with the MoreComing flag set, then any of those answers are equally likely to work.

Reply:
So in this case there's a browse going on for `_matterd._udp`.  For each instance name that finds (in this case just `D49A7571551947A4`) we do a resolve to find the SRV and TXT records.  In this case it looks like we are getting back two results, with the same SRV record (same hostname and port, which is not surprising since there is only one `_matterd._udp` advertiser involved) but different TXT records, on the same interface.  @Abhayakara is this situation expected?  I can see this perhaps happening if we got the first result, but it had the "more results coming" flag (so we leave the resolve going) and then an update comes in.  But it doesn't sound like that's what's going on here...

And it's not clear what "try both" would mean here: it's the same entity on the network (same hostname/port), but the metadata about it in the TXT records is different for the two TXT records.  @cliffamzn I assume that you are using that DN TXT record as part of some sort of UI or something and it's ending up showing data that's out of sync with what's actually being advertised?

Reply:
@bzbarsky-apple correct it was some UI that was showing data that is out of sync with what is actually being advertised. In the case of the example app it was always only showing the 1st entry and missing the 2nd.

After digging into this a bit more, i realize that our advertiser did not have this change:
https://github.com/project-chip/connectedhomeip/pull/23933

So it wasn't expiring the old entry and instead keeping both instances around. After getting that pulled in, it looks like it actually helped resolve the problem.

Actually it kind of makes for an interesting problem though because that means that devices running the v1.0 version of the matter SDK might do the same thing (since this fix came in after it was released), so commissioners looking at "old" clients will probably run into a similar issue.

It _appears_ that this solution has been working since the observed behavior seems to suggest some sort of linear ordering based on time it was added. That is purely anecdotal though based on very limited testing.